### ISSUE TITLE
Replace Java environment vars with fully-native

### ISSUE THREAD

**Issue by headius** [jruby/jruby#6248]
While investigating the test failures reported in #6171 I ran across a CGI test that fails:

```ruby
req = Net::HTTP::Get.new("/webrick.cgi/%A4%DB%A4%B2/%A4%DB%A4%B2")
http.request(req){|res|
  assert_equal("/\xA4\xDB\xA4\xB2/\xA4\xDB\xA4\xB2", res.body, log.call)
}
```

This URL gets parsed out and passed to a CGI subprocess via an environment variable, but because it does not decode as valid UTF-8 it's not possible for JRuby to either set or get that variable without mangling the original bytes.

Unfortunately the root cause is Java's own `System.getenv`, which promises a map from String to String and therefore must be able to decode environment variables to characters.

The only way we can support this use case, where differently-encoded text or undecodable bytes are stored in an environment variable, would be to bypass `System.getenv` and go straight to the native `getenv` function. Since this is a much larger fix than I wanted to attempt for #6171, I'm filing this issue and will expand WEBrick's own test exclusion to also exclude the "java" platform.

**Comment by headius** [comment]
This might be a good introductory issue for anyone looking to get into JRuby, since the ENV logic is fairly self-contained:

https://github.com/jruby/jruby/blob/6c0f3370f5b550cfd857483e4c82fefff8daf27f/core/src/main/java/org/jruby/RubyGlobal.java#L397-L404

**Comment by headius** [comment]
@deivid-rodriguez I added a comment on #40 about expanding that test's exclusion to also exclude the "java" platform. If you'd rather I spin up my own PR just let me know.

**Comment by imlakshay08** [comment]
Hey @headius , I would like to work on this issue. Could please assign me this ?

**Comment by headius** [comment]
@imlakshay08 I can't assign you without making you part of one of the teams, and we don't do that until someone shows a pattern of contributing. Go right ahead and work on it if you like!

**Comment by imlakshay08** [comment]
Oh...okay. Alright.

**Comment by CufeHaco** [comment]
Is this still open?  Im just browsing to see what I can help with.  I can always do prototype work for the team to look over and decide what you want to use, and how i can contribute.

**Comment by enebo** [comment]
@CufeHaco yeah.   Go for it.   It looks like the last volunteer did not finish it.

**Comment by CufeHaco** [comment]
@enebo I didn't forget this one, I lost my bookmark for it.  Ill jump on it while I wait for @headius on the PR's he requested.

If you have any specific older PRs you would like me to work on alongside with Charles, feel free to tag me in a new thread.  It will help me learn java.  Im a stay at home dad and wanting to switch my career back to programming, so I'm doing this in the mindset of a full time job and using my Electrical background and ethos into my work.
